x-pack/plugins/observability_solution/dataset_quality/public/hooks/use_degraded_docs_chart.tsx

```diff
@@ -0,0 +1,130 @@
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback, useState } from 'react';
+import { Action } from '@kbn/ui-actions-plugin/public';
+
+import { i18n } from '@kbn/i18n';
+import { DEFAULT_LOGS_DATA_VIEW } from '../../common/constants';
+import { indexNameToDataStreamParts } from '../../common/utils';
+import { getLensAttributes } from '../components/flyout/degraded_docs_trend/lens_attributes';
+import { useCreateDataView } from './use_create_dataview';
+import { useLinkToLogsExplorer } from './use_link_to_logs_explorer';
+import { useDatasetQualityFlyout } from './use_dataset_quality_flyout';
+import { useKibanaContextForPlugin } from '../utils';
+
+const exploreDataInLogsExplorerText = i18n.translate(
+  'xpack.datasetQuality.flyoutChartExploreDataInLogsExplorerText',
+  {
+    defaultMessage: 'Explore data in Logs Explorer',
+  }
+);
+
+const openInLensText = i18n.translate('xpack.datasetQuality.flyoutChartOpenInLensText', {
+  defaultMessage: 'Open in Lens',
+});
+
+const ACTION_EXPLORE_IN_LOGS_EXPLORER = 'ACTION_EXPLORE_IN_LOGS_EXPLORER';
+const ACTION_OPEN_IN_LENS = 'ACTION_OPEN_IN_LENS';
+
+interface DegradedDocsChartDeps {
+  dataStream?: string;
+}
+
+export const useDegradedDocsChart = ({ dataStream }: DegradedDocsChartDeps) => {
+  const {
+    services: { lens },
+  } = useKibanaContextForPlugin();
+  const { dataStreamStat, timeRange } = useDatasetQualityFlyout();
+
+  const [isChartLoading, setIsChartLoading] = useState<boolean | undefined>(undefined);
+  const [attributes, setAttributes] = useState<ReturnType<typeof getLensAttributes> | undefined>(
+    undefined
+  );
+
+  const datasetTypeIndexPattern = dataStream
+    ? `${indexNameToDataStreamParts(dataStream).type}-*-*`
+    : undefined;
+  const { dataView } = useCreateDataView({
+    indexPatternString: datasetTypeIndexPattern ?? DEFAULT_LOGS_DATA_VIEW,
+  });
+  const filterQuery = `_index: ${dataStream ?? 'match-none'}`;
+
+  const handleChartLoading = (isLoading: boolean) => {
+    setIsChartLoading(isLoading);
+  };
+
+  const openInLensCallback = useCallback(() => {
+    if (attributes) {
+      lens.navigateToPrefilledEditor({
+        id: '',
+        timeRange,
+        attributes,
+      });
+    }
+  }, [lens, attributes, timeRange]);
+
+  const getOpenInLensAction = (): Action => {
+    return {
+      id: ACTION_OPEN_IN_LENS,
+      type: 'link',
+      order: 17,
+      getDisplayName(): string {
+        return openInLensText;
+      },
+      getIconType(): string {
+        return 'visArea';
+      },
+      async isCompatible(): Promise<boolean> {
+        return true;
+      },
+      async execute(): Promise<void> {
+        return openInLensCallback();
+      },
+    };
+  };
+
+  const logsExplorerLinkProps = useLinkToLogsExplorer({
```

Comment:
`useLinkToLogsExplorer` needs a `timeRange` param as it currently uses the table's time range. If flyout has a different time range, the links to Lens and Explorer will land on different time ranges.

Reply:
The table and flyout have their timeRange properties as timeRange be changed for the flyout separately. So, if the time ranges is changed on the flyout, the `useLinkToLogsExplorer ` would still use the table's time range state.

Currently, if flyout is opened and time range is changed on the flyout, the link to Lens will use the correct time range (flyout's one) whereas the link to Explorer will use the table's time range.

Reply:
Exactly!